src/main/java/org/mockito/internal/stubbing/answers/ReturnsArgumentAt.java

```diff
@@ -51,24 +51,26 @@ public ReturnsArgumentAt(int wantedArgumentPosition) {
     }
 
     @Override
-    public Object answer(InvocationOnMock invocation) throws Throwable {
-        int argumentPosition = inferWantedArgumentPosition(invocation);
-        validateIndexWithinInvocationRange(invocation, argumentPosition);
+    public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+        Invocation invocation = (Invocation) invocationOnMock;
 
-        if (wantedArgIndexIsVarargAndSameTypeAsReturnType(
-                invocation.getMethod(), argumentPosition)) {
+        if (wantedArgIndexIsVarargAndSameTypeAsReturnType(invocation)) {
             // answer raw vararg array argument
-            return ((Invocation) invocation).getRawArguments()[argumentPosition];
+            int rawArgumentPosition = inferWantedRawArgumentPosition(invocation);
+            return invocation.getRawArguments()[rawArgumentPosition];
         }
 
+        int argumentPosition = inferWantedArgumentPosition(invocation);
+        validateIndexWithinInvocationRange(invocation, argumentPosition);
+
         // answer expanded argument at wanted position
         return invocation.getArgument(argumentPosition);
     }
 
     @Override
     public void validateFor(InvocationOnMock invocation) {
         int argumentPosition = inferWantedArgumentPosition(invocation);
-        validateIndexWithinInvocationRange(invocation, argumentPosition);
+        validateIndexWithinTheoreticalInvocationRange(invocation, argumentPosition);
```

Comment:
When this is invoked, both `getArguments()` and `getRawArguments()` return the same, which is the number of _raw_ arguments.

I've renamed to `theoretical` because at this point in the code the actual parameters are not known, so we're only asking is its theoretically possible that the argument position is valid.